xray/commands/audit/generic/jas/applicabilityscanner.go

```diff
@@ -0,0 +1,336 @@
+package jas
+
+import (
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/jfrog/jfrog-client-go/xray/services"
+	"github.com/owenrumney/go-sarif/v2/sarif"
+	"gopkg.in/yaml.v2"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	ApplicableStringValue    = "Applicable"
+	NotApplicableStringValue = "Not Applicable"
+	UndeterminedStringValue  = "Undetermined"
+)
+
+const (
+	applicabilityScanCommand        = "ca"
+	applicabilityScanType           = "analyze-applicability"
+	applicabilityScanFailureMessage = "failed to run applicability scan: "
+)
+
+var (
+	analyzerManagerExecuter                  AnalyzerManager = &analyzerManager{}
+	technologiesEligibleForApplicabilityScan                 = []coreutils.Technology{coreutils.Npm, coreutils.Pip,
+		coreutils.Poetry, coreutils.Pipenv, coreutils.Pypi}
+)
+
+type ExtendedScanResults struct {
+	XrayResults                 []services.ScanResponse
+	ApplicabilityScannerResults map[string]string
+	EntitledForJas              bool
+}
+
+func (e *ExtendedScanResults) GetXrayScanResults() []services.ScanResponse {
+	return e.XrayResults
+}
+
+func GetExtendedScanResults(results []services.ScanResponse, dependencyTrees []*services.GraphNode,
+	serverDetails *config.ServerDetails) (*ExtendedScanResults, error) {
+	err := createAnalyzerManagerLogDir()
+	if err != nil {
+		return nil, err
+	}
+	applicabilityScanManager, err := NewApplicabilityScanManager(results, dependencyTrees, serverDetails)
+	if err != nil {
+		log.Info(applicabilityScanFailureMessage + err.Error())
+		return nil, err
+	}
+	shouldRun, err := applicabilityScanManager.shouldRun()
+	if err != nil {
+		log.Info(applicabilityScanFailureMessage + err.Error())
+		return nil, err
+	}
+	if !shouldRun {
+		log.Info("user not entitled for jas, didnt execute applicability scan")
+		return &ExtendedScanResults{XrayResults: results, ApplicabilityScannerResults: nil, EntitledForJas: false}, nil
+	}
+	entitledForJas, err := applicabilityScanManager.Run()
+	if !entitledForJas {
+		log.Info("got not entitled error from analyzer manager")
```

Comment:
log.Info("got not entitled error from analyzer manager")
-->
log.Debug("The current user is not entitled for the Advanced Security package.")